vote/src/vote_account.rs

```diff
@@ -131,8 +131,7 @@ impl VoteAccount {
     pub fn bls_pubkey(&self) -> Option<BLSPubkey> {
         let bls_pubkey_compressed = self.0.vote_state_view.bls_pubkey_compressed()?;
         let bls_pubkey_compressed_bytes = BLSPubkeyCompressed(bls_pubkey_compressed);
-        let bls_pubkey = BLSPubkey::try_from(bls_pubkey_compressed_bytes).unwrap();
-        Some(bls_pubkey)
+        BLSPubkey::try_from(bls_pubkey_compressed_bytes).ok()
```

Comment:
What do you feel about just returning self.0.vote_state_view.bls_pubkey_compressed()?
I think we can handle uncompress and related errors in bls_pubkey_map.

That way vote account doesn't have to verify whether its data is valid, and we can handle:
1. There is no bls pubkey written
2. The written bls pubkey is wrong
differently. I feel this will be useful for the ledger tool which updates BLS pubkey later.

Reply:
I'm fine having this API just return `self.0.vote_state_view.bls_pubkey_compressed()`

It would feel a little strange pushing decompression onto all consumers like `epoch_stakes`. I'm thinking we need a decompression API somewhere. We could translate different cases (no BLS pubkey, invalid BLS pubkey, etc.) into `Err` codes

Reply:
I'm looking at the caller of bls_pubkey() in Alpenglow repo, there are two main callers:
1. epoch_stakes.rs, it should cache uncompressed bls_pubkey in bls_pubkey_to_rank_map.
2. vote_util where we compare bls_pubkey in vote account with the one we have, we can cache our own compressed key so we just do a comparison with compressed key.

So we should also be good without a decompression API inside vote_account.

Reply:
IIUC, you're saying we never need to decompress, right?

Reply:
Nah, we do need to decompress in bls_pubkey_to_rank_map, I believe all the verifications use decompressed pubkey. But we only need to do that once to all vote accounts when new EpochStakes is generated.

Reply:
right, so where do we want the decompression API to live?

Reply:
yes, that would functionally work.

But it feels strange pushing decompression onto the consumer (`epoch_stakes` in this case). It seems like an implementation detail that vote accounts compress the pubkey to save on space. I don't think we should push that complexity onto all potential consumers.

It probably makes sense to expose both APIs from `vote_account`:
1. "I want a compressed pubkey because I'm space conscious"
2. "Just give me the normal pubkey"